lib/ui/window/pointer_data.cc

```diff
@@ -12,4 +12,26 @@ static constexpr int kPointerDataFieldCount = 19;
 static_assert(sizeof(PointerData) == sizeof(int64_t) * kPointerDataFieldCount,
               "PointerData has the wrong size");
 
+void PointerData::Clear() {
+  time_stamp = 0;
+  pointer = 0;
+  change = Change::kCancel;
```

Comment:
How about making Change::kCancel and DeviceKind::kTouch be zero in the enum and then in the struct constructor, bzero the memory to the sizeof the struct.  That way, you won't have to remember to explicitly call Clear() and won't need to explicitly zero out the value if more struct members are added. If you don't want to change the enum, you could still zero it out in the constructor and then set it to whatever you want. The Clear seems like something one might forget to call and end up with garbage values. The struct has no union types, so the bzero should work fine.